pyNN/common/procedural_api.py

```diff
@@ -82,7 +82,6 @@ def record(variables, source, filename, sampling_interval=None, annotations=None
         source.record(variables, to_file=filename, sampling_interval=sampling_interval)
         if annotations:
             source.annotate(**annotations)
-        simulator.state.write_on_end.append((source, variables, filename))
```

Comment:
@apdavison : As discussed, the particular sub-test has been commented out, pending re-implementation with pyNN.mock